src/components/oauth/OAuthCodeHandler.js

```diff
@@ -0,0 +1,100 @@
+/**
+ * @author sarahr
+ */
+
+import React, { useCallback, useEffect, useState } from "react";
+import { useRouter } from "next/router";
+import { useQuery } from "react-query";
+import { build } from "@cyverse-de/ui-lib";
+
+import {
+    APP_CATEGORIES_QUERY_KEY,
+    getPrivateCategories,
+} from "serviceFacades/apps";
+import {
+    OAUTH_CALLBACK_QUERY_KEY,
+    doOAuthCallback,
+} from "serviceFacades/oauth";
+
+import OAuthErrorHandler from "./OAuthErrorHandler";
+import OAuthLinkingMessage from "./OAuthLinkingMessage";
+import ids from "./ids";
+import { getListingPath } from "components/apps/utils";
+import systemId from "components/models/systemId";
+import { useUserProfile } from "contexts/userProfile";
+
+const ERR_SERVICE = "general_service_error";
+
+function OAuthCodeHandler(props) {
+    const { baseId, apiName, code, stateId } = props;
+    const handlerId = build(baseId, ids.OAUTH_CODE_HANDLER);
+    const router = useRouter();
+    const [userProfile] = useUserProfile();
+    const [categoryQueryKey, setCategoryQueryKey] = useState([
+        APP_CATEGORIES_QUERY_KEY,
+        userProfile?.id,
+    ]);
+    const [listingUrl, setListingUrl] = useState(
+        getListingPath(null, null, null, null, null, null, false)
```

Comment:
I used a general app listing URL as the default. That seemed to be the most reasonable thing to do if we can't determine the URL for HPC app listings.